std/io/bufio.ts

```diff
@@ -706,5 +706,15 @@ export async function* readStringDelim(
 export async function* readLines(
   reader: Reader,
 ): AsyncIterableIterator<string> {
-  yield* readStringDelim(reader, "\n");
+  const encoder = new TextEncoder();
+  const decoder = new TextDecoder();
+  for await (let chunk of readDelim(reader, encoder.encode("\n"))) {
+    // Finding a CR at the end of the line is evidence of a
+    // "\r\n" at the end of the line. The "\r" part should be
+    // removed too.
+    if (chunk[chunk.length - 1] == CR) {
+      chunk = chunk.slice(0, chunk.length - 1);
+    }
+    yield decoder.decode(chunk);
+  }
```

Comment:
Actually, couldn't it be written like this?
```suggestion
  for await (let chunk of readStringDelim(reader, "\n")) {
    // Finding a CR at the end of the line is evidence of a
    // "\r\n" at the end of the line. The "\r" part should be
    // removed too.
    if (chunk.endsWith("\r")) {
      chunk = chunk.slice(0, -1);
    }
    yield chunk;
  }
```

Reply:
Yes, you are right. At the moment I assumed working with `Uint8Array` was preferable to `string`  for performance reasons, but honesly I can't think of any reason why. Your suggestion is certainly cleaner and easier to understand, thank you!